logging/google/cloud/logging/logger.py

```diff
@@ -436,29 +462,34 @@ def log_proto(self, message, labels=None, insert_id=None, severity=None,
         """
         self.entries.append(
             ('proto', message, labels, insert_id, severity, http_request,
-             timestamp))
+             timestamp, resource))
 
-    def commit(self, client=None):
+    def commit(self, client=None, resource=None):
```

Comment:
So the reason to do it this way is because when you write entries you can leave some entries without resources and override them at the top level. I just wanted to expose that level of the API, maybe not strictly necessary.